### PR TITLE
For #800 - Adds metrics for the default search engine

### PR DESCRIPTION
**Pull request by boek**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
* Missing roboelectric :(
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- Does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by boek** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) **What questions will you answer with this data?**

- What the landscape of which search engines users set as their default

2) **Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?**

- We want to make sure that users are using the correct URL and other relevant search engine information

3) **What alternative methods did you consider to answer these questions? Why were they not sufficient?**

* N/A (These are baseline metrics)

4) **Can current instrumentation answer these questions?**

* Currently no, as these are some of the first metrics we're recording

5) **List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data c](https://wiki.mozilla.org/Firefox/Data_Collection)[ollection ](https://wiki.mozilla.org/Firefox/Data_Collection)[categories](https://wiki.mozilla.org/Firefox/Data_Collection) on the found on the Mozilla wiki.**

* All data is Category 1.

6) **How long will this data be collected?**

Until 03/01/2020

7) **What populations will you measure?**

* All release, beta, and nightly users with telemetry enabled.

8) **Please provide a general description of how you will analyze this data.**

* Glean / Amplitude

9) **Where do you intend to share the results of your analysis?**

* Only on Glean, Amplitude and with mobile teams.

**Comment by boek** [comment]
CI issue will get fixed by #1604

File: app/metrics.yaml [hunk 1]
```diff
@@ -326,4 +326,42 @@ metrics:
...
+    bugs:
+    - 800
+    data_reviews:
+    - TBD
+    notification_emails:
+    - fenix-core@mozilla.com
+    expires: "2020-03-01"
+  name:
+    type: string
+    description: The name of the search engine
```
Comment (liuche) [inline app/metrics.yaml]: I'm concerned about the case where Fenix starts adding custom search engines.
A few thoughts here - can we add some comments, like for Search Shortcuts in #1194? Also, I'll pre-emptively file a bug for adding custom search engines, and link to both this bug and #1194 as places that we need to be careful about not tracking.

File: app/metrics.yaml [hunk 2]
```diff
@@ -326,4 +326,42 @@ metrics:
...
+    bugs:
+    - 800
+    data_reviews:
+    - TBD
+    notification_emails:
+    - fenix-core@mozilla.com
+    expires: "2020-03-01"
+  submission_url:
+    type: string
+    description: The full url we use to query the serach engine
```
Comment (liuche) [inline app/metrics.yaml]: Nit: typo, and can you clarify that this without the search term?

**Comment by liuche** [comment]
Filed #1607 as a reminder when custom search engines are added that the telemetry needs to be updated.

**Comment by boek** [comment]
6) **How long will this data be collected?**

Until ~03/01/2020~ 09/01/2019

**Pull request merged**